.github/workflows/docker-server.yml

```diff
@@ -55,6 +56,31 @@ jobs:
           build-args: |
             GITHUB_SHA_SHORT=${{ steps.build_args.outputs.github_sha_short }}
 
+      - name: Prepare CICD args
+        id: cicd_args
+        run: |
+          commit_author=$(echo ${{ github.event.pusher.name }} | tr -d \"\')
+          commit_message=$(echo ${{ github.event.head_commit.message }} | tr -d \"\')
```

Comment:
Either I don't understand how these substitutions work, or this is still wrong:

What if the commit message contains `); rm -rf /`? That string just gets substituted in here without any escaping and screws everything up?

I'm not seeing anything in the docs about what escaping is performed when substituting or how to do it manually. This seems like an enormous oversight so maybe I'm missing something?

Reply:
Alright, here: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections
They recommend using environment variables. If you do that properly (with toJSON and all the proper shell quoting) I think it should work.
i.e.
```
env:
  COMMIT_MESSAGE: ${{ toJSON(github.event.head_commit.message) }}
run: |
  curl ... -d '{..., "commit_message": '"$COMMIT_MESSAGE"', ...}'
```
note that you end the single quotes around the json, start a double-quote to avoid word-splitting the env var, substitute the env var, end the double quote, then resume the single quote for the rest of the json.